rosidl_runtime_c/include/rosidl_runtime_c/string_functions.h

```diff
@@ -55,6 +55,24 @@ ROSIDL_GENERATOR_C_PUBLIC
 void
 rosidl_runtime_c__String__fini(rosidl_runtime_c__String * str);
 
+/// Copy rosidl_runtime_c__String structure content.
+/**
+ * This functions performs a deep copy, as opposed to the shallow copy that
+ * plain assignment yields.
+ *
+ * \param[out] input a pointer to a rosidl_runtime_c__String structure
+ *   to copy from.
+ * \param[out] output a pointer to an initialized rosidl_runtime_c__String
+ *   structure to copy into.
+ * \return true if successful, false if either pointer is null or memory
+ *   allocation fails.
+ */
+ROSIDL_GENERATOR_C_PUBLIC
+bool
+rosidl_runtime_c__String__copy(
+  const rosidl_runtime_c__String * input,
+  rosidl_runtime_c__String * output);
```

Comment:
Does it make sense to declare `rosidl_runtime_c__String * const output)` so we declare that output pointer itself is not being modified?

Reply:
Same for `input`. If so, there is a good bunch of input/output constness to adjust in the PR.

Reply:
We could, but I *think* that might be slightly confusing to people. We're passing the pointer by value, so there's not at all to be gained from marking it `const`.